app/src/main/java/org/mifos/mobile/ui/fragments/BeneficiaryApplicationFragment.kt

```diff
@@ -157,54 +129,71 @@ class BeneficiaryApplicationFragment : BaseFragment(), BeneficiaryApplicationVie
+                tilBeneficiaryName.editText?.setText(beneficiary?.name)
+                tilTransferLimit.editText?.setText(beneficiary?.transferLimit.toString())
+            }
         } else if (beneficiaryState == BeneficiaryState.CREATE_QR) {
-            accountTypeField?.setText(beneficiary?.accountType?.value!!, false)
-            tilAccountNumber?.editText?.setText(beneficiary?.accountNumber)
-            tilOfficeName?.editText?.setText(beneficiary?.officeName)
+            with(binding) {
+                accountTypeField.setText(beneficiary?.accountType?.value!!, false)
+                tilAccountNumber.editText?.setText(beneficiary?.accountNumber)
+                tilOfficeName.editText?.setText(beneficiary?.officeName)
+            }
         }
     }
 
     /**
      * Used for submitting a new or updating beneficiary application
      */
-    @OnClick(R.id.btn_beneficiary_submit)
-    fun submitBeneficiary() {
-        tilAccountNumber?.isErrorEnabled = false
-        tilOfficeName?.isErrorEnabled = false
-        tilTransferLimit?.isErrorEnabled = false
-        tilBeneficiaryName?.isErrorEnabled = false
-        if (accountTypeId == -1) {
-            Toaster.show(rootView, getString(R.string.choose_account_type))
-            return
-        } else if (tilAccountNumber?.editText?.text.toString().trim { it <= ' ' } == "") {
-            tilAccountNumber?.error = getString(R.string.enter_account_number)
-            return
-        } else if (tilOfficeName?.editText?.text.toString().trim { it <= ' ' } == "") {
-            tilOfficeName?.error = getString(R.string.enter_office_name)
-            return
-        } else if (tilTransferLimit?.editText?.text.toString() == "") {
-            tilTransferLimit?.error = getString(R.string.enter_transfer_limit)
-            return
-        } else if (tilTransferLimit?.editText?.text.toString() == ".") {
-            tilTransferLimit?.error = getString(R.string.invalid_amount)
-            return
-        } else if (tilTransferLimit?.editText?.text.toString().matches("^0*".toRegex())) {
-            tilTransferLimit?.error = getString(R.string.amount_greater_than_zero)
-            return
-        } else if (tilBeneficiaryName?.editText?.text.toString().trim { it <= ' ' } == "") {
-            tilBeneficiaryName?.error = getString(R.string.enter_beneficiary_name)
-            return
+    private fun submitBeneficiary() {
+        with(binding) {
+            tilAccountNumber.isErrorEnabled = false
+            tilOfficeName.isErrorEnabled = false
+            tilTransferLimit.isErrorEnabled = false
+            tilBeneficiaryName.isErrorEnabled = false
+        }
+        with(binding) {
+            when {
+                accountTypeId == -1 -> {
+                    Toaster.show(root, getString(R.string.choose_account_type))
+                    return
+                }
+                tilAccountNumber.editText?.text.toString().trim { it <= ' ' } == "" -> {
```

Comment:
One more thing...!  😊 Isn't `tilAccountNumber.editText?.text.toString().trim { it <= ' ' } == "" ` equivalent to `tilAccountNumber.editText?.text.isNullOrBlank()`? 🤔  What do you think?

Reply:
yes @jawidMuhammadi it does!! . I will add another commit to replace its usage throughout the file

Reply:
done @jawidMuhammadi !!